extensions/mssql/src/objectManagement/commands.ts

```diff
@@ -455,6 +459,40 @@ async function handleDropDatabase(context: azdata.ObjectExplorerContext, service
 	}
 }
 
+async function handleRestoreDatabase(context: azdata.ObjectExplorerContext, service: IObjectManagementService): Promise<void> {
+	const connectionUri = await getConnectionUri(context);
+	if (!connectionUri) {
+		return;
+	}
+	try {
+		const parentUrn = await getParentUrn(context);
+		const options: ObjectManagementDialogOptions = {
+			connectionUri: connectionUri,
+			isNewObject: true,
+			database: context.connectionProfile!.databaseName!,
+			objectType: ObjectManagement.NodeType.Database,
+			objectName: '',
+			parentUrn: parentUrn,
+			objectExplorerContext: context
+		};
+		const dialog = new RestoreDatabaseDialog(service, options);
+		const startTime = Date.now();
+		await dialog.open();
+		TelemetryReporter.sendTelemetryEvent(TelemetryActions.OpenRestoreDatabaseDialog, {
+			objectType: ObjectManagement.NodeType.Database
+		}, {
+			elapsedTimeMs: Date.now() - startTime
+		});
+	}
+	catch (err) {
+		TelemetryReporter.createErrorEvent2(ObjectManagementViewName, TelemetryActions.OpenRestoreDatabaseDialog, err).withAdditionalProperties({
+			objectType: context.nodeInfo!.nodeType
+		}).send();
+		console.error(err);
+		await vscode.window.showErrorMessage(objectManagementLoc.OpenRestoreDatabaseDialogError(getErrorMessage(err)));
```

Comment:
You generally shouldn't await show*Message calls unless you want to respond to the user since that'll block until the user interacts with the dialog (i.e. closes it). It doesn't technically matter since this is all in a callback and nothing else happens afterwards - but it's better to just change it now to avoid potential issues later if someone adds stuff after this call.